internal/clients/tgw.go

```diff
@@ -0,0 +1,56 @@
+package clients
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/client/network_service"
+	networkmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-network/preview/2020-09-07/models"
+	sharedmodels "github.com/hashicorp/hcp-sdk-go/clients/cloud-shared/v1/models"
+)
+
+// GetTGWAttachmentByID gets a TGW attachment by its ID, hvnID, and location
+func GetTGWAttachmentByID(ctx context.Context, client *Client, tgwAttachmentID string, hvnID string, loc *sharedmodels.HashicorpCloudLocationLocation) (*networkmodels.HashicorpCloudNetwork20200907TGWAttachment, error) {
+	getTGWAttachmentParams := network_service.NewGetTGWAttachmentParams()
+	getTGWAttachmentParams.ID = tgwAttachmentID
+	getTGWAttachmentParams.HvnID = hvnID
+	getTGWAttachmentParams.HvnLocationOrganizationID = loc.OrganizationID
+	getTGWAttachmentParams.HvnLocationProjectID = loc.ProjectID
+	getTGWAttachmentResponse, err := client.Network.GetTGWAttachment(getTGWAttachmentParams, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	return getTGWAttachmentResponse.Payload.TgwAttachment, nil
+}
+
+// WaitForTGWAttachmentState will poll the GET TGW attachment endpoint until
+// it is in the specified state, ctx is canceled, or an error occurs. This
+// is required because AWS can return a newly created TGW attachment before
+// it is ready to be used.
+func WaitForTGWAttachmentState(ctx context.Context, client *Client, tgwAttachmentID string, hvnID string, loc *sharedmodels.HashicorpCloudLocationLocation, state string) (*networkmodels.HashicorpCloudNetwork20200907TGWAttachment, error) {
+	for {
+		tgwAtt, err := GetTGWAttachmentByID(ctx, client, tgwAttachmentID, hvnID, loc)
+		if err != nil {
+			return nil, fmt.Errorf("unable to retrieve Transit gateway attachment (%s): %v", tgwAttachmentID, err)
```

Comment:
Does `Gateway` need to be capitalized in these error messages?

Reply:
I'm honestly not sure here. Our capitalization guide from design doesn't seem to explicitly call out a case like this: https://docs.google.com/document/d/1ABNkYIp6aLwPYNpuYOFcDFBFCcnH_4ePRaLP-gD_tKE/edit

@joshklekamp Do you have an opinion here on how we capitalize `Transit gateway attachment` within our error messages and logs?

Reply:
It should be sentence case unless we add AWS to the beginning.

Reply:
or Amazon 😄

Reply:
AWS itself calls it "Transit Gateway"

Reply:
Verified with Josh that we should use sentence case here (so `Transit gateway attachment` at the beginning of a sentence, but `transit gateway attachment` in the middle of a sentence). There are two exceptions where we refer to the resource/data source by the full name in their top-level description, and in those cases we use `AWS Transit Gateway Attachment`. I've updated the error messages and descriptions to reflect the correct capitalization.